### PR TITLE
Fixes Grid-Overlay with node Compatibility

### LINKED ISSUE #1006: Grid Overlay error
### If we change the input values for GridOverlay, The grid, disappears.

ezgif com-video-to-gif(6)

### Please show us where to look

https://beta.sequencer.publiclab.org

### PR DESCRIPTION
**Pull request by aashna27**
Fixes #1006
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `npm test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/is-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by HarshKhandeparkar** [comment]
@aashna27 can't thie module be switched to use PixelManipulation? Using canvas isn't required here(i think). It will just make it incompatible with node (we can use puppeteer but it will just take more memory).

**Comment by jywarren** [comment]
Actually i think this was @Harshithpabbati's module implementation originally... i might be wrong? @Harshithpabbati is that right? Thanks, all!

**Comment by HarshKhandeparkar** [comment]
It was harshith's module.

**Comment by Harshithpabbati** [comment]
Wait for a min I will check it.

**Comment by aashna27** [comment]
> @aashna27 can't thie module be switched to use PixelManipulation? Using canvas isn't required here(i think). It will just make it incompatible with node (we can use puppeteer but it will just take more memory).

Yeah valid point , I ll do it using pixelmanipulation, in few days.

**Comment by HarshKhandeparkar** [comment]
So can the merging of this PR be put on hold? If yes, add a WIP in the title and add a label.

**Comment by aashna27** [comment]
I have done this using pixel manipulation, as suggested by Harsh and it works for node as well now.

File: src/modules/GridOverlay/GridOverlay.js [hunk 1]
```diff
@@ -1,36 +1,28 @@
...
-        var p=2;
-        function drawBoard(){
-            for (var x = 0; x <= canvas.width; x+=options.x) {
-                ctx.moveTo(0.5 + x + p, p);
-                ctx.lineTo(0.5 + x + p, canvas.height + p);
-            }
-            for (var y = 0; y <= canvas.height; y+=options.y) {
-                ctx.moveTo(p, 0.5 + y + p);
-                ctx.lineTo(canvas.width + p, 0.5 + y + p);
+        for(var x = 0; x <= pixels.shape[0]; x+=options.x){
```
Comment (HarshKhandeparkar) [inline src/modules/GridOverlay/GridOverlay.js]:
```suggestion
        for(var x = 0; x < pixels.shape[0]; x+=options.x){
```

File: src/modules/GridOverlay/GridOverlay.js [hunk 2]
```diff
@@ -1,36 +1,28 @@
...
+            for(var y = 0 ; y <= pixels.shape[1]; y++){
```
Comment (HarshKhandeparkar) [inline src/modules/GridOverlay/GridOverlay.js]:
```suggestion
            for(var y = 0 ; y < pixels.shape[1]; y++){
```

File: src/modules/GridOverlay/GridOverlay.js [hunk 3]
```diff
@@ -1,36 +1,28 @@
...
+                pixels.set(x, y, 2, color[2]);
+                pixels.set(x, y, 3, color[3]);
             }
-            ctx.strokeStyle = options.color;
-            ctx.stroke();
         }
-        
-        drawBoard();
+    
+        for(var y = 0; y <= pixels.shape[1]; y+=options.y){
```
Comment (HarshKhandeparkar) [inline src/modules/GridOverlay/GridOverlay.js]:
```suggestion
        for(var y = 0; y < pixels.shape[1]; y+=options.y){
```

File: src/modules/GridOverlay/GridOverlay.js [hunk 4]
```diff
@@ -1,36 +1,28 @@
...
+            for(var x = 0 ; x <= pixels.shape[0]; x++){
```
Comment (HarshKhandeparkar) [inline src/modules/GridOverlay/GridOverlay.js]:
```suggestion
            for(var x = 0 ; x < pixels.shape[0]; x++){
```

Comment (aashna27) [reply src/modules/GridOverlay/GridOverlay.js]: @HarshKhandeparkar width and height isn't anything out of bound.

Comment (HarshKhandeparkar) [reply src/modules/GridOverlay/GridOverlay.js]: How can the x be equal to the width? It is 0 indexed so it has to stop at width - 1, right? Am I saying something wrong?

Comment (aashna27) [reply src/modules/GridOverlay/GridOverlay.js]: No its perfect, valid point.

**Comment by HarshKhandeparkar** [comment]
@aashna27 could you please do me a favor?

**Comment by aashna27** [comment]
Yes, please let me know.

**Comment by HarshKhandeparkar** [comment]
What OS do you use?

**Comment by HarshKhandeparkar** [comment]
I wanted someone to test #1038 locally.

**Comment by aashna27** [comment]
I use windows, yeah I ll do it but I will need sometime, I have 2 exams scheduled tomorrow and I don't want to be doing anything in haste.

**Comment by HarshKhandeparkar** [comment]
Np. Thanks

**Comment by rexagod** [comment]
@HarshKhandeparkar 

```sh
 % npm test

> image-sequencer@3.3.0 test /mnt/c/Users/Rexagod/Desktop/is
> TEST=true istanbul cover tape test/core/*.js test/core/ui/user-interface.js test/core/modules/canvas-resize.js test/core/modules/QR.js test/core/modules/crop.js | tap-spec; browserify --global-transform [ babelify --plugins [ @babel/plugin-syntax-object-rest-spread @babel/plugin-proposal-object-rest-spread ] ] test/core/modules/image-sequencer.js test/core/modules/chain.js test/core/modules/meta-modules.js test/core/modules/replace.js test/core/modules/import-export.js test/core/modules/run.js test/core/modules/dynamic-imports.js test/core/util/parse-input.js test/core/modules/benchmark.js| tape-run --render="tap-spec"


  Output directory is correctly generated

    ✔ Access the created dir

  Notify function works for all three UIs

    ✔ Default notify() produces correct output
    ✔ Html UI contains notify function

  Load canvas-resize module

    ✔ Canvas resize module is getting loaded

  Check Options

    ✔ Options are correct
    ✔ Options are correct

  canvas-resize module works correctly

    ✔ should be equal
    ✔ should be equal

  Load add QR module

    ✔ Add-QR module is getting loaded.

  Check Options

    ✔ Options are correct

  Add QR module works correctly

    ✔ should be equal

  Crop module loads correctly

    ✔ Crop module loaded

  Crop module loads with correct options

    ✔ options loaded correctly

  Crop module works correctly

    ✔ should be equal
=============================================================================
Writing coverage object [/mnt/c/Users/Rexagod/Desktop/is/coverage/coverage.json]
Writing coverage reports at [/mnt/c/Users/Rexagod/Desktop/is/coverage]
=============================================================================
    =============================== Coverage summary ===============================
    Statements   : 35.11% ( 697/1985 )
    Branches     : 27.48% ( 177/644 )
    Functions    : 20.63% ( 72/349 )
    Lines        : 35.75% ( 686/1919 )
    ================================================================================


  total:     14
  passing:   14
  duration:  1m 14.9s


[BABEL] Note: The code generator has deoptimised the styling of /mnt/c/Users/Rexagod/Desktop/is/node_modules/lodash/lodash.js as it exceeds the max of 500KB.
npm ERR! Test failed.  See above for more details.
```

**Comment by HarshKhandeparkar** [comment]
@rexagod I think you posted this on a different pr

**Comment by rexagod** [comment]
Yeah I posted this in #1038 as well.

**Comment by aashna27** [comment]
Adding ready-to -merge here.

**Comment by jywarren** [comment]
Great, thanks so much!!!!!

**Comment by aashna27** [comment]
Always a learning opportunity. 👐

**Pull request merged**